src/main/java/hillel/spring/doctors/DoctorRestRepository.java

```diff
@@ -0,0 +1,73 @@
+package hillel.spring.doctors;
+
+
+import hillel.spring.doctors.model.Doctor;
+import lombok.AllArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+@Repository
+@AllArgsConstructor
+public class DoctorRestRepository {
+
+    private final List<Doctor> doctors = new CopyOnWriteArrayList<>();
+    private static int id;
+
+    @Bean
+    public AtomicInteger getAtomicInteger() {
```

Comment:
`Repository` is not usual place for methods returning `@Bean`s.
And integer (even atomic) is not a good candidate for a bean. So it is better to have it as a field rather than as a bean.

Reply:
will be removed to config class